### PR TITLE
AK+LibWeb: Start porting Web API events to new String

### PR DESCRIPTION
**Pull request by kennethmyhra**
Includes a conversion function to convert from FlyString to DeprecatedString.

### CODE REVIEW

File: Userland/Libraries/LibWeb/CSS/MediaQueryListEvent.h [hunk 1]
```diff
@@ -6,32 +6,33 @@
...
 #pragma once
 
+#include <AK/FlyString.h>
 #include <LibWeb/DOM/Event.h>
 
 namespace Web::CSS {
 
 struct MediaQueryListEventInit : public DOM::EventInit {
-    DeprecatedString media { "" };
+    String media { String {} };
```
Comment (linusg) [inline Userland/Libraries/LibWeb/CSS/MediaQueryListEvent.h:15]: You can omit this: `String media;`

File: Userland/Libraries/LibWeb/HTML/CloseEvent.h [hunk 2]
```diff
@@ -7,37 +7,38 @@
...
+#include <AK/FlyString.h>
 #include <LibWeb/DOM/Event.h>
 
 namespace Web::HTML {
 
 struct CloseEventInit : public DOM::EventInit {
     bool was_clean { false };
     u16 code { 0 };
-    DeprecatedString reason { "" };
+    String reason { String {} };
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/CloseEvent.h:18]: Likewise

File: Userland/Libraries/LibWeb/HTML/ErrorEvent.h [hunk 3]
```diff
@@ -6,14 +6,15 @@
...
 #include <LibWeb/DOM/Event.h>
 
 namespace Web::HTML {
 
 // https://html.spec.whatwg.org/multipage/webappapis.html#erroreventinit
 struct ErrorEventInit : public DOM::EventInit {
-    DeprecatedString message { "" };
-    DeprecatedString filename { "" }; // FIXME: This should be a USVString.
+    String message { String {} };
+    String filename { String {} }; // FIXME: This should be a USVString.
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/ErrorEvent.h:17]: And everywhere else

**Comment by kennethmyhra** [comment]
Done!

**Review by linusg — APPROVED** [review]

**Pull request merged**